src/txn/txn.c

```diff
@@ -138,6 +138,42 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     }
 }
 
+/*
+ * __wt_txn_user_active --
+ *     Check whether there are any running user transactions.
+ */
+bool
+__wt_txn_user_active(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_TXN_GLOBAL *txn_global;
+    WT_TXN_SHARED *s;
+    uint32_t i, session_cnt;
+    bool txn_active;
+
+    conn = S2C(session);
+    txn_global = &conn->txn_global;
+    txn_active = true;
+
+    /* We're going to scan the table in exclusive mode: wait for the write lock. */
+    __wt_writelock(session, &txn_global->rwlock);
+
+    /* Walk the array of concurrent transactions. */
+    WT_ORDERED_READ(session_cnt, conn->session_cnt);
+    WT_STAT_CONN_INCR(session, txn_walk_sessions);
+    for (i = 0, s = txn_global->txn_shared_list; i < session_cnt; i++, s++) {
+        WT_STAT_CONN_INCR(session, txn_sessions_walked);
+        /* If session has a valid transaction id, there is at least one active transaction. */
+        if (s->id != WT_TXN_NONE && !F_ISSET(session, WT_SESSION_INTERNAL))
+            goto done;
```

Comment:
We can get rid of this `goto` and simplify the method.
1. Init `txn_active` with `false` at the binning of the method. At the moment it's initialized with `true`.
2. Modify end loop check `i < session_cnt && !txn_active`
3. Replace `if` block with this `txn_active = (s->id == WT_TXN_NONE || F_ISSET(session, WT_SESSION_INTERNAL));`
4. Remove `txn_active = false;` at line 171

Reply:
I have reworked checks and got rid of `goto`.